### PR TITLE
Pass back phpBB login failure reason to DP login

### PR DESCRIPTION
**Pull request by cpeel**
Pass back the underlying phpBB failure reason to the DP login page.
Route the user to the forums if their maximum number of login attempts
were exceeded so they can log in through the CAPTCHA.

This is available for testing in the [phpbb3-login-improvements](https://www.pgdp.org/~cpeel/c.branch/phpbb3-login-improvements/) sandbox.

### CODE REVIEW

File: accounts/login_failure.php [hunk 1]
```diff
@@ -26,6 +26,8 @@
     'no_username'      => _("You did not supply a username."),
     'no_password'      => _("You did not supply a password."),
     'auth_failure'     => _("Unable to authenticate. The username/password may be incorrect or your account may be locked."),
+    'unknown_failure'  => sprintf(_("An unexpected failure occurred, please contact the <a href='%s'>site manager</a>."), "mailto:$site_manager_email_addr"),
```
Comment (srjfoo) [inline accounts/login_failure.php]: I'd probably say `a` site manager instead of `the` site manager

Comment (cpeel) [reply accounts/login_failure.php]: I'm fine either way, this is using the same wording as the other paragraph at the bottom of the file.

Comment (srjfoo) [reply accounts/login_failure.php]: Just grammatical pickiness -- `the` is specific, and implies that there's only one.

File: accounts/login_failure.php [hunk 2]
```diff
@@ -55,7 +57,7 @@
...
+echo "<p>" . sprintf( _("If all of this fails, contact a <a href='%s'>site manager</a>."), "mailto:$site_manager_email_addr") . "</p>";
+echo "<p>" . sprintf( _("Note: If you have just registered, you will need to wait for the welcome mail to arrive to your mailbox. Once it does, please click the activation link to complete the registration (this is to prevent others from signing you up to the site without your knowledge). If you have waited for an hour or so and have still not received any mail from us (please check any spam filters!), it is likely that you misentered your email-address. Please contact a <a href='%s'>site manager</a> to solve the problem."), "mailto:$site_manager_email_addr") . "</p>";
```
Comment (srjfoo) [inline accounts/login_failure.php]: This para probably ought to have a separate $testing option, also. It gives the false impression that users will actually receive an activation email. Possibly something along the lines of (further wordsmithing welcome):

> sprintf( _("Note: This is a testing site, and is not set up to send emails. If you have just registered, the page generated should have had an activation link on the page for you to copy and paste into a new browser tab. If you closed that page and can no longer access the information on it, please contact a <a href='%s'>site manager</a> to solve the problem."), "mailto:$site_manager_email_addr")

Comment (cpeel) [reply accounts/login_failure.php]: Good call and good wording. Sandbox updated.

**Pull request merged**